setup/docker/utils.go

```diff
@@ -0,0 +1,38 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+)
+
+func LoadConfig(path string) (*Config, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var config *Config
+	if err := json.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("Failed to unmarshal %s: %v", path, err)
+	}
+
+	if config.Auths == nil {
+		config.Auths = map[string]Auth{}
+	}
+
+	return config, nil
+}
+
+func WriteConfig(path string, config *Config) error {
+	data, err := json.MarshalIndent(config, "", "    ")
+	if err != nil {
+		return fmt.Errorf("Failed to marshal docker config: %v", err)
+	}
+
+	if err := ioutil.WriteFile(path, data, 0644); err != nil {
```

Comment:
consider changing the permissions here to `0600`. There's no need for group and everyone to have read or write access to this file.